### DISCUSSION TITLE
DataHub -> Data Objects Importer   update not possible

### DISCUSSION THREAD

**Discussion by ismetsener** [pimcore/pimcore#9822] [Q&A / Support]
Hi, 
I installed DataHub (Data Objects Importer)
I imported  a lot of objects  (via .csv file)

Now I try to update some fields of the Imported Objects
for example there are 4 attribute fields,
how can I  update only one of the attributes, other 3 are not changed

I tried with "Processing Settings"   " Delta Check"

but I get error:
` ERROR     [DATA-IMPORTER] Error processing element: Exception: Duplicate full path [ /Office/Test/filename of the object ] - cannot save object in /opt/pimcore/vendor/pimcore/pimcore/models/DataObject/AbstractObject.php:831`

I get this error because the filename  is not changed.

**Comment by fashxp** [comment]
Hi, 
you have to setup the loading strategy in the resolver settings. This strategy is responsible for checking, if data objects are already present and loads them instead of creating new objects (with the same path). Details see also [here](https://pimcore.com/docs/data-importer/current/Configuration/Resolver_Settings.html#page_Element-Loading). 

For details on the delta check, see [here](https://pimcore.com/docs/data-importer/current/Configuration/Processing_Settings.html#page_Delta-Check). It checks, if data in import file has changed since last import. So, I'm not sure, if it does exactly what you need.

**Comment by ismetsener** [comment]
I did exactly what was mentioned in 
`https://pimcore.com/docs/data-importer/current/Configuration/index.html`

Resolver Settings  and Processing Settings  are configured as in document
the first import was worked perfect
after the update was not OK. the system says that it is duplicate value

**Reply by fashxp** [reply]
It is obvious, that the first import works if objects are not there yet (then it is not possible to have duplicate paths 😉.
The loading strategy becomes relevant after the first import when data objects are already there and eventually need to be loaded. 

So either the loading strategy is not setup correctly, or there is a bug in the system ... in both cases it would be great if you can dig a bit deeper, see whats wrong and provide a fix if there is something wrong in the data importer. 

Thank you very much...

**Comment by ismetsener** [comment]
I will try to solve :)

but I read in the documents:
`The delta check is based on hashes and compares the import data only. It does not consider any other changes of the data object if self, that might occur in the meantime between two imports.`

what do they mean by this?

**Reply by fashxp** [reply]
that means, that the hash is built based on the import-data and also checked against import data and does not consider any changes in the data object directly. 

for example:

- import 1: `myname;mydescription` results in hash `123`
- import 2: `myNEWname;mydescription` results in hash `234` --> delta check detects difference, import is done. 
- import 3: `myNEWname;mydescription` results in hash `234` --> delta check detects **no** difference, import is skipped. 
- attribute is changed in data object directly 
- import 4: `myNEWname;mydescription` results in hash `234` --> delta check detects **no** difference, import is skipped - even though the data object has changed in the meantime.

**Comment by ismetsener** [comment]
ok thanks now I understand

the problem is that the system says that it is duplictae.

for example:

IMPORT first time
object 1)
object name:  product_abc
name:Venus
color: red
date: 2021-11-01

object 2)
object name:  product_xyz
name:Mars
color: blue
date: 2021-11-01

NOW UPDATE  (everything same, but i changed the date)
object 1)
object name:  product_abc
name:Venus
color: red
date: 2021-05-05

object 2)
object name:  product_xyz
name:Mars
color: blue
date: 2021-05-05

the system starts the process and comes with an error:
```
Error processing element: Exception: Duplicate full path    /path/product_abc 
Error processing element: Exception: Duplicate full path    /path/product_xyz

```

the object name is not changed, only what changed is the date.
the loading strategy is  "id"  
data source index: 0   (object name)
I did also other combinations, but all with an error


in this way I have to delete all the objects and import again.
or i have to change the object name and import/update again, but then it is double, with old object name and object file name

**Reply by fashxp** [reply]
so your second import file contains the IDs of the objects that were created during the first import in its first column?
Otherwise the loading strategy ID cannot work, as it loads the data objects based on their ID - and if it doesn't find any, it creates new objects...

**Reply by ismetsener** [reply]
Yes, the second import (update) contains the ID's that were created during the first import.
I expect that becuase the ID exists,  that it only updates the data that is changed,
this is what I expect.

but the system does not update, is this an error?

**Reply by fashxp** [reply]
just tested ... resolving by id worked for me, existing data objects were updated. 

![image](https://user-images.githubusercontent.com/8792145/126953597-537354e4-7e73-43f6-8ef6-05a46c1511f1.png)

maybe you can debug that a bit further on your end.

**Reply by ismetsener** [reply]
can you show me also your mappings and processings settings?
it looks (green part) that you did not map all the fileds?

**Reply by fashxp** [reply]
these settings should not matter for loading existing objects. the only relevant settings you see in the screen above.

**Reply by ismetsener** [reply]
with configuration of what I have it does NOT update, but creates always new objects.
and if object exists the system says  Duplicate value.

the loading strategy is id (object name) this exists, so it should update and not try to create new object

**Reply by fashxp** [reply]
can you please share extracts of the two files you are importing

**Reply by ismetsener** [reply]
PROBLEM SOLVED 

I tested again it WORKS  :)


there was a  "id"  field/record in the datahub, i used this as filename  :(
now the "id"  field/record in datahub is same as "id" in object field
